### ISSUE TITLE
-1004:SyntaxError: Where clause in optional match is not supported. near `kyquyen_congdong|e_dangkyquyen'. How to alternate OPTIONAL MATCH with WHERE in version 3.4 ? 

### ISSUE THREAD

**Issue by son2408** [vesoft-inc/nebula#5368]
This' s my graph.
```
MATCH(v1:dangkyquyen) -[e1:e_dangkyquyen_tinhhinhdangky|e_dangkyquyen_vochong|e_dangkyquyen_hogiadinh]-> (v2) WHERE v1.dangkyquyen.NodeId IN ['de9b68d1-6327-4c38-b689-d2a027ed2a96']
OPTIONAL MATCH(v2) -[e2:e_hogiadinh_canhan|e_vochong_canhan|e_tochuc_canhan|e_tinhhinhdangky_canhan]-> (v3) WHERE v3.canhan.isLastest == 1
return v1.dangkyquyen.NodeId,v2.tinhhinhdangky.NodeId,v2.vochong.NodeId,v2.hogiadinh.NodeId,v3.canhan.NodeId
```

**Comment by Sophie-Xie** [comment]
This issue is the same as https://github.com/vesoft-inc/nebula/issues/5362.

**Comment by son2408** [comment]
@Sophie-Xie this use-case, I want to return v1, v2 if v3 null. Instead of use option match, what's other ?

**Comment by son2408** [comment]
@wey-gu

**Comment by xtcyclist** [comment]
```
MATCH (v1:dangkyquyen) -[e1:e_dangkyquyen_tinhhinhdangky|e_dangkyquyen_vochong|e_dangkyquyen_hogiadinh]->(v2)-[e2:e_hogiadinh_canhan|e_vochong_canhan|e_tochuc_canhan|e_tinhhinhdangky_canhan]->(v3) WHERE v1.dangkyquyen.NodeId IN ['de9b68d1-6327-4c38-b689-d2a027ed2a96'] and v3.canhan.isLastest == 1 \
return v1.dangkyquyen.NodeId,v2.tinhhinhdangky.NodeId,v2.vochong.NodeId,v2.hogiadinh.NodeId,v3.canhan.NodeId \
UNION \
MATCH(v1:dangkyquyen) -[e1:e_dangkyquyen_tinhhinhdangky|e_dangkyquyen_vochong|e_dangkyquyen_hogiadinh]->(v2) WHERE v1.dangkyquyen.NodeId IN ['de9b68d1-6327-4c38-b689-d2a027ed2a96'] \
return v1.dangkyquyen.NodeId,v2.tinhhinhdangky.NodeId,v2.vochong.NodeId,v2.hogiadinh.NodeId,null
```

Maybe something like this. Not equivalent though.

**Comment by son2408** [comment]
> MATCH (v1:dangkyquyen) -[e1:e_dangkyquyen_tinhhinhdangky|e_dangkyquyen_vochong|e_dangkyquyen_hogiadinh]->(v2)-[e2:e_hogiadinh_canhan|e_vochong_canhan|e_tochuc_canhan|e_tinhhinhdangky_canhan]->(v3) WHERE v1.dangkyquyen.NodeId IN ['de9b68d1-6327-4c38-b689-d2a027ed2a96'] and v3.canhan.isLastest == 1 \
> return v1.dangkyquyen.NodeId,v2.tinhhinhdangky.NodeId,v2.vochong.NodeId,v2.hogiadinh.NodeId,v3.canhan.NodeId \
> UNION \
> MATCH(v1:dangkyquyen) -[e1:e_dangkyquyen_tinhhinhdangky|e_dangkyquyen_vochong|e_dangkyquyen_hogiadinh]->(v2) WHERE v1.dangkyquyen.NodeId IN ['de9b68d1-6327-4c38-b689-d2a027ed2a96'] \
> return v1.dangkyquyen.NodeId,v2.tinhhinhdangky.NodeId,v2.vochong.NodeId,v2.hogiadinh.NodeId,null

-1009:SemanticError: different column names to UNION/INTERSECT/MINUS are not supported. Cannot return result of v3. Set operators only support same return of MATCHES

**Comment by son2408** [comment]
I want MATCH1 UNION MATCH2 UNION MATCH3 UNION MATCH4 and return all result although null

```
ex: MATCH(a) -[e: a_b]-> (b)
	MATCH(b) -[e: b_c]-> (c)
	MATCH(c) -[e: c_d]-> (d)
	RETURN a,b,c,d
```

I wan't graph MATCH(a)-[e:a_b]->(b)-[e2:b_c]-> (c)-[e3: c_d]->(d). Because if one of graph is null then all null.
The other way graph a->b->c->d still return result if null  @xtcyclist  @wey-gu

**Comment by xtcyclist** [comment]
In 3.3, we have discovered that the `OPTINAL MATCH` with `WHERE` produces wrong results. And, it takes too much (refactoring a very large porition of the code base) to fix it. We tried to find an affordable solution but didn't find one. In the end, we have to ban the syntax in 3.4. 

With this syntax banned, there is no guarantee that an OPTIONAL MATCH with WHERE could be rewritten into some other queries and be executed correctly.

For your case:
```
MATCH(v1) -[e:e_12] ->(v2)
MATCH(v2) -[e:e_23] ->(v3)
MATCH(v3) -[e:e_34] ->(v4)
```

I recommend you to split it into multiple queries and union the results in your own app.
